src/api_server/src/lib.rs

```diff
@@ -264,7 +264,17 @@ impl ApiServer {
             Ok(ParsedRequest::Sync(vmm_action)) => {
                 self.serve_vmm_action_request(vmm_action, request_processing_start_us)
             }
-            Ok(ParsedRequest::GetInstanceInfo) => self.get_instance_info(),
+            Ok(ParsedRequest::GetInstanceInfo) => {
+                if self.instance_info.state.as_str() == "Not started" {
+                    if let Ok(VmmData::State(VmState::Running)) =
+                        self.send_vmm_action(Box::new(VmmAction::GetVmState))
+                    {
+                        self.instance_info.state = "Running".to_string();
+                    }
+                }
+
+                self.get_instance_info()
+            }
```

Comment:
The fact that `GetInstanceInfo` is directly handled by the ApiServer is an optimization that made sense when ApiServer was the only way to configure and start a uVM. I think it unnecessarily complicates things now (since we introduces `--config-file`).

While this PR gets the job done nicely, I would propose we simplify it by just moving state information to `Vmm/Controller` and uniformly handle the requests there regardless of how uVM was configured.

wdyt?

Reply:
I think the whole `instance_info`, looking at it https://github.com/firecracker-microvm/firecracker/blob/9e94be71f4a079bf8fafd8f3c801750e744d930f/src/api_server/swagger/firecracker.yaml#L800 it doesn't have any apiserver-specific info in it, so I see no reason to split its implementation.

I would have it work as any other api call: api server turns it into a RPC request to the controller, the controller handles it, comes back with a response, api server packages it to http and sends the response